### PR TITLE
[.NET] Custom Teams Endpoint (CTE) GA

### PR DESCRIPTION
**Pull request by AikoBB**
# Description

Preparing CTE for the GA:

- Added mandatory parameters(appId, userId) to getTokenForTeamsUser
- Added GA API version to ServiceVersion

## Implementation TODOs
- [x] Update Identity SDK to point to new CTE path
  - [x] regenerate implementation from a new swagger spec
  - [x] update recordings
- [x] changes regarding to ACS verification of App ID & UserId
  - [x] regenerate implementation from a new swagger spec - https://github.com/Azure/azure-rest-api-specs/pull/17930
  - [x] update client implementation
  - [x] update & add tests
  - [x] update recordings
  - [x] update README & CHANGELOG
  - [x] update tag and require path in swagger/README.md
-[x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Comment by petrsvihlik** [comment]
@AikoBB if you by any chance read this, could you please give me contributor rights to your branch?

File: sdk/communication/Azure.Communication.Identity/samples/Sample1_CommunicationIdentityClient.md [hunk 1]
```diff
@@ -63,12 +63,12 @@ Console.WriteLine($"Token: {token}");
...
 
-## Exchange an AAD access token of a Teams User for a Communication Identity access token
+## Exchange an Azure AD access token of a Teams User for a Communication Identity access token
 
-The `CommunicationIdentityClient` can be used to exchange an AAD access token of a Teams user for a new Communication Identity access token with a matching expiration time.
+The `CommunicationIdentityClient` can be used to exchange an Azure AD access token of a Teams user for a new Communication Identity access token with a matching expiration time.
 
 ```C# Snippet:GetTokenForTeamsUser
-Response<AccessToken> tokenResponse = client.GetTokenForTeamsUser(teamsToken);
+Response<AccessToken> tokenResponse = client.GetTokenForTeamsUser(options);
```
Comment (petrsvihlik) [inline sdk/communication/Azure.Communication.Identity/samples/Sample1_CommunicationIdentityClient.md:71]: Can we extend the snippet (`Sample1_CommunicationIdentityClient.cs`) to demonstrate what params need to be passed into the options bag? Or extend the description to explain it in the text? (Similarly to the sample right above it --> Generate a user token)

File: sdk/communication/Azure.Communication.Identity/README.md [hunk 2]
```diff
@@ -114,10 +114,10 @@ Response revokeResponse = await client.RevokeTokensAsync(user);
 Response deleteResponse = await client.DeleteUserAsync(user);
 ```
 
-### Exchanging AAD access token of a Teams User for a Communication Identity access token
+### Exchanging Azure AD access token of a Teams User for a Communication Identity access token
```
Comment (petrsvihlik) [inline sdk/communication/Azure.Communication.Identity/README.md:117]: Similarly to the Sample1_CommunicationIdentityClient.md, I'd recommend adding a better description of how to utilize the method and how to initialize the options bag.

File: sdk/communication/Azure.Communication.Identity/samples/Sample1_CommunicationIdentityClientAsync.md [hunk 3]
```diff
@@ -65,12 +65,12 @@ Console.WriteLine($"Token: {token}");
 Console.WriteLine($"Expires On: {expiresOn}");
 ```
 
-## Exchange an AAD access token of a Teams User for a Communication Identity access token
+## Exchange an Azure AD access token of a Teams User for a Communication Identity access token
```
Comment (petrsvihlik) [inline sdk/communication/Azure.Communication.Identity/samples/Sample1_CommunicationIdentityClientAsync.md:68]: See my comments for Sample1_CommunicationIdentityClient.md

File: sdk/communication/Azure.Communication.Identity/tests/CommunicationIdentityClient/CommunicationIdentityClientLiveTests.cs [hunk 4]
```diff
@@ -164,51 +193,113 @@ public async Task GetTokenForTeamsUserWithEmptyTokenShouldThrow()
...
             {
                 CommunicationIdentityClient client = CreateClientWithConnectionString();
-                Response<AccessToken> tokenResponse = await client.GetTokenForTeamsUserAsync(TestEnvironment.CommunicationExpiredTeamsToken);
+                Response<AccessToken> tokenResponse = await client.GetTokenForTeamsUserAsync(new GetTokenForTeamsUserOptions(CTEOptions.TeamsUserAadToken, CTEOptions.UserObjectId, CTEOptions.UserObjectId));
             }
             catch (RequestFailedException ex)
             {
                 Assert.NotNull(ex.Message);
-                Assert.True(ex.Message.Contains("401"));
+                Assert.True(ex.Message.Contains("400"));
```
Comment (petrsvihlik) [inline sdk/communication/Azure.Communication.Identity/tests/CommunicationIdentityClient/CommunicationIdentityClientLiveTests.cs:254]: btw how come this has changed from 401 to 400 @paveldostalms @AikoBB  ? is it correct?

**Review by petrsvihlik — CHANGES_REQUESTED** [review]

File: sdk/communication/Azure.Communication.Identity/tests/CommunicationIdentityClient/CommunicationIdentityClientLiveTests.cs (hunk 4 above)
Comment (AikoBB) [reply sdk/communication/Azure.Communication.Identity/tests/CommunicationIdentityClient/CommunicationIdentityClientLiveTests.cs:254]: yep, for the new API version, the service responds with 400 status code for invalid params. Correct me if I am wrong @paveldostalms

Comment (paveldostalms) [reply sdk/communication/Azure.Communication.Identity/tests/CommunicationIdentityClient/CommunicationIdentityClientLiveTests.cs:254]: Yes, invalid clientId or userId is causing 400 as requested by PM. Invalid token continue to return 401.

File: sdk/communication/Azure.Communication.Identity/tests/Infrastructure/CommunicationIdentityClientLiveTestBase.cs [hunk 5]
```diff
@@ -51,14 +59,10 @@ private CommunicationIdentityClientOptions CreateIdentityClientOptionsWithCorrel
             return InstrumentClientOptions(communicationIdentityClientOptions);
         }
 
-        protected async Task<string> generateTeamsToken()
+        protected async Task<GetTokenForTeamsUserOptions> createTeamsUserParams()
```
Comment (petrsvihlik) [inline sdk/communication/Azure.Communication.Identity/tests/Infrastructure/CommunicationIdentityClientLiveTestBase.cs:62]:
```suggestion
        protected async Task<GetTokenForTeamsUserOptions> CreateTeamsUserParams()
```

File: sdk/communication/Azure.Communication.Identity/tests/CommunicationIdentityClient/CommunicationIdentityClientLiveTests.cs [hunk 6]
```diff
@@ -21,12 +21,17 @@ namespace Azure.Communication.Identity.Tests
...
+        /// Options used to exchange an AAD access token of a Teams user for a new Communication Identity access token.
+        /// </summary>
+        internal GetTokenForTeamsUserOptions CTEOptions;
         /// <summary>
         /// Initializes a new instance of the <see cref="CommunicationIdentityClient"/> class.
         /// </summary>
         /// <param name="isAsync">A flag used by the Azure Core Test Framework to differentiate between tests for asynchronous and synchronous methods.</param>
         public CommunicationIdentityClientLiveTests(bool isAsync) : base(isAsync)
         {
+            CTEOptions = createTeamsUserParams().Result;
```
Comment (petrsvihlik) [inline sdk/communication/Azure.Communication.Identity/tests/CommunicationIdentityClient/CommunicationIdentityClientLiveTests.cs:34]: Calling `.Result` is not a good practice. I'd avoid setting the CTEOptions in the constructor and rather used `[OneTimeSetup]` from [NUnit3](https://docs.nunit.org/articles/nunit/writing-tests/attributes/onetimesetup.html) which can be async.

**Review by petrsvihlik — CHANGES_REQUESTED** [review]

**Review by petrsvihlik — APPROVED** [review]

**Pull request merged**